lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -173,6 +173,23 @@ public void Close(byte[] message)
                 _currentStatus.SetException(TaskClientCodeException.Create(
                     TaskId, ContextId, "Error during Close().", e));
             }
+            finally
+            {
+                Logger.Log(Level.Info, "TaskRuntime.Close, finally._userTask: " + _userTask);
```

Comment:
You should validate that the `ITask` thread is finished before calling `Dispose`.